enable postgres as the backend of `vault`

pivotal-bin-ju opened the pull request:
Fixes: https://github.com/concourse/prod/issues/44

**This PR can work with the forked repo: https://github.com/concourse/vault-helm**
We are going to draft a PR for that later.

I created a new vault - `vault-nci` for test, resilience and migration purpose.
the steps are as below:
- `ln -s vault vault-nci`
- `make creds-vault-nci`
- `make deploy-vault-nci`

## You can review the PR, but DO NOT Merge the PR until the data migration is done.

pivotal-bin-ju committed:
enable postgres as the backend of `vault`

- terraform: add a cloudsql/postgres
- chart: use postgres as the backend

Signed-off-by: Bin Ju <bju@pivotal.io>

cirocosta commented on deployments/with-creds/vault/README.md:
```diff
@@ -1,3 +1,29 @@
+1. `vault-nci` and `vault` are alive at the same time so far. `vault` may decommission in the future after `vault-nci` is all set.
+1. How to deploy `vault-nci`
+    1. `git clone https://github.com/concourse/vault-helm` to the `charts` (hush-house/deployments/with-creds/vault/charts) directory.
+    1 `ln -s vault vault-nci` 
```
hmmmm this seems like a transitory step that has been used during development 🤔  

do you plan to have the deployment named after `vault-nci` or `vault? I guess the former? 

(just because if the latter , then we wouldn't really need https://github.com/concourse/hush-house/pull/102 as the name of the service and namespace wouldn't change)

if you feel like the transition is good (checked via the scripts that ensure that we got the data transferred well), I'd say these steps aren't needed - we should have in this repo just the final configuration that reflect the environment 🤔

pivotal-bin-ju replied:
I've manually back-upped the `/vault/data` again to make the process more safe if we only keep the `vault` instance. Anyway the `PVC` is still attached to `vault` we could switch back to the FS backend anytime.

I agree.

cirocosta reviewed (commented):
It seems pretty good 😁  I'm just concerned about the way that things are represented here - If you're  confident that we got all set up correctly, in which case `vault` is removed in favor of `vault-nci` (the postgres-backed one) then I think we should have in this commit the removal of the old one, and the addition of the new (rather than the linking, etc) 😁

pivotal-bin-ju committed:
keep maintain one instance of vault

after the data between `vault` and `vault-nci`, the old vault is going
to decommission, the new `vault-nci` becomes `vault`. After the change
user can only see the instance `vault`, that would be simple for the
later maintenance.

Signed-off-by: Bin Ju <bju@pivotal.io>
